geode-core/src/main/java/org/apache/geode/internal/cache/wan/AbstractGatewaySender.java

```diff
@@ -846,12 +849,29 @@ public void distribute(EnumListenerEvent operation, EntryEventImpl event,
 
     // If this gateway is not running, return
     if (!isRunning()) {
+      if (this.isPrimary()) {
+        // clone event without offheap
+        EntryEventImpl droppedEvent = new EntryEventImpl(event, false);
+        droppedEvent.disallowOffHeapValues();
+
+        if (this.eventProcessor != null) {
```

Comment:
I might be missing something here but it seems there is a chance that we never process the tmpDroppedEvents list.  Is that ok?

Imagine distribute being called before event process is set, we add to tmpDroppedEvents.  The next time distribute is called, the sender is running and so we don't his this if block.

Reply:
That's a good catch. I will fix.